nixos/modules/services/hardware/logiops.nix

```diff
@@ -0,0 +1,164 @@
+      type = types.lines;
+      default = "";
+      example = ''
+        devices: ({
+          name: "Wireless Mouse MX Master";
+          smartshift:
+          {
+            on: true;
+            threshold: 30;
+            torque: 50;
+          };
+          hiresscroll:
+          {
+            hires: true;
+            invert: false;
+            target: false;
+          };
+          dpi: 1000;
+
+          buttons: (
+            {
+              cid: 0xc3;
+              action =
+              {
+                type: "Gestures";
+                gestures: (
+                  {
+                    direction: "Up";
+                    mode: "OnRelease";
+                    action =
+                    {
+                      type: "Keypress";
+                      keys: ["KEY_UP"];
+                    };
+                  },
+                  {
+                    direction: "Down";
+                    mode: "OnRelease";
+                    action =
+                    {
+                      type: "Keypress";
+                      keys: ["KEY_DOWN"];
+                    };
+                  },
+                  {
+                    direction: "Left";
+                    mode: "OnRelease";
+                    action =
+                    {
+                      type: "CycleDPI";
+                      dpis: [400, 600, 800, 1000, 1200, 1400, 1600];
+                    };
+                  },
+                  {
+                    direction: "Right";
+                    mode: "OnRelease";
+                    action =
+                    {
+                      type = "ToggleSmartshift";
+                    }
+                  },
+                  {
+                    direction: "None"
+                    mode: "NoPress"
+                  }
+                );
+              };
+            },
+            {
+              cid: 0xc4;
+              action =
+              {
+                type: "Keypress";
+                keys: ["KEY_A"];
+              };
+            }
+          );
+        });
+      '';
+      description = mdDoc ''
+        Extra configuration that is appended to the end of the file.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.etc."logid.cfg".text = ''
+      ${cfg.extraConfig}
+    '';
+
+    systemd.services.logiops = {
+      description = "Logitech Configuration Daemon";
+      documentation = [ "https://github.com/PixlOne/logiops" ];
+
+      wantedBy = [ "graphical.target" ];
+
+      startLimitIntervalSec = 0;
+      after = [ "multi-user.target" ];
+      wants = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStart = "${pkgs.logiops}/bin/logid";
+        Restart = "always";
+        # Ensures that the hid_logitech_hidpp module is loaded before logid starts
+        # https://github.com/PixlOne/logiops/issues/269#issuecomment-1012608838
+        ExecStartPre = "${pkgs.kmod}/bin/modprobe hid_logitech_hidpp || :";
```

Comment:
An alternative solution is to use a `udev` rule to restart the service after the kernel module has loaded:
```
  # Add a udev rule to restart `logiops` when the mouse is connected
  # https://github.com/PixlOne/logiops/issues/156#issuecomment-1887054527
  services.udev.extraRules = ''
    ACTION=="add", KERNEL=="hid_logitech_hidpp", RUN+="${pkgs.systemd}/bin/systemctl --no-block restart logiops.service"
  '';
```